### DISCUSSION TITLE
Attach a trait to equipment

### DISCUSSION THREAD

**Discussion by Marv93** [richardwilkes/gcs#663] [Ideas]
I tried to build the following a piece of equipment that when equipped gives the character a leveled trait. First I tried building the trait via equipment features but that doesn't support levels. So I built the trait and then tried attaching it to the equipment which also doesn't work. My workaround is to just manually equip someone with both, the equipment and the trait, but it could be more elegant, especially when there is more than one trait attached to a piece of equipment.

One way would be: Add a new feature for equipment: "Gives the trait" and then select one of a list of traits.

Doing this would enable the creation of equipment that can give any number of traits to a character and it would enable different pieces of equipment giving the same traits (or already existing traits in general) without having to rebuild them for each piece of equipment.

**Comment by richardwilkes** [comment]
You should create an issue for this. It's a significant amount of work, though (largely because of the need to handle a ton of edge cases that this will create in the UI), so it is something that I'm not likely to do soon as I have other things I'd rather do right now.
